BNote/src/presentation/widgets/filebrowser.php

```diff
@@ -280,64 +257,152 @@ private function writeFolderContent() {
 			}
 			$dirname = str_replace("/", " > ", $dirname);
 			Writing::h3($dirname, "filebrowser_folder_header");
-			
+
 			// show table with files
-			$content = $this->getFilesFromFolder($this->root . $this->path);
-			?>
-			<div class="filebrowser_filepanel">
-				<?php
-				$this->writeFolderContentItem($content, "folder");
-				$this->writeFolderContentItem($content, "file");
-				?>
-			</div>
-			<?php
+			$content = $this->getFilesFromFolder($this->path);
+			echo '			<div class="filebrowser_filepanel">';
+
+			// show folders as list items
+			for($i = 1; $i < count($content); $i++) {
+				$item = $content[$i];
+				if($item["directory"] === true)
+					$this->writeFolderContentListItem($item);
+			}
+
+			// show files as list items
+			for($i = 1; $i < count($content); $i++) {
+				$item = $content[$i];
+				if(($item["directory"] === false) && ($item["tile"] === false))
+					$this->writeFolderContentListItem($item);
+			}
+
+			// show files as tile items
+			for($i = 1; $i < count($content); $i++) {
+				$item = $content[$i];
+				if(($item["directory"] === false) && ($item["tile"] === true))
+					$this->writeFolderContentTileItem($item);
+			}
+			echo '			</div>';
 		}
 	}
-	
-	private function writeFolderContentItem($content, $type) {
-		for($i = 1; $i < count($content); $i++) {
-			$item = $content[$i];
-			
-			if($type == "folder" && $item["directory"] === false) continue;
-			else if($type == "file" && $item["directory"] === true) continue;
-			
+
+	private function writeFolderContentListItem($item) {
+			/*
+			 * All infomation to be shown is copied into simple variables that can
+			 * be used in a HEREDOC string.
+			 */
```

Comment:
This comment isn't helping much

Reply:
What do you suggest? Should I remove the comment or should I point out here that array members like $item["show"] can't appear inside a HEREDOC string, but simple variables like $link are allowed?

Reply:
I think this comment is not needed.